test/TesterInternal/EventStoreTests/CommonEventStoreTests.cs

```diff
@@ -0,0 +1,340 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TestExtensions;
+using Xunit.Abstractions;
+using Orleans.Runtime.Storage;
+using Orleans.Storage;
+using Orleans.Runtime;
+using Xunit;
+using Orleans;
+using System.Threading;
+
+namespace Tester.EventStorageTests
+{
+    /// <summary>
+    /// Common tests for event stores.
+    /// To use these tests, derive from this class, and specify 
+    /// the store to test via the abstract property.
+    /// All tests create fresh GUIDs, so no cleaning is necessary in between tests.
+    /// </summary>
+    public abstract class CommonEventStoreTests
+    {
+        protected abstract IEventStorage StoreUnderTest { get; }
+
+        // if it is not possible to directly derive from this class, call this to run all 
+        public async Task RunAll()
+        {
+            await One1();
+            await One2();
+            await One3();
+            await Five1();
+            await Five2();
+            await More1();
+            await More2();
+            await Empty1();
+            await Empty2();
+            await Empty3();
+            await Delete1();
+            await Delete2();
+            await Delete3();
+            await Delete4();
```

Comment:
Please use more descriptive test names, especially if the plan is to be reused by new provider implementations. We already have very cryptic tests in our codebase that we are trying to rename as we review and re-understand them, but let's not add to that. It's fine if the test name is super long.